ketohub/spiders/raw_content_spider.py

```diff
@@ -62,10 +80,23 @@ def download_recipe_contents(self, response):
         [download_root]/YYYYMMDD/hhmmssZ/[source_domain]/[relative_url]/
          """
         # Build path for scraped files
-        filepath = os.path.join(self._filepath_prefix, response.url[12:])
+        if not self._filepath_prefix:
+            self._set_download_root()
+
+        filepath = os.path.join(self._filepath_prefix,
+                                self._format_recipe_key(response.url))
 
         # Write response body to file
-        _write_to_file(filepath, response.text.encode('utf8'))
+        _write_to_file(filepath, 'index.html', response.text.encode('utf8'))
+
+        # Write url to metadata file
+        _write_to_file(filepath, 'metadata.json',
+                       json.dumps(
+                           {
+                               'url': response.url
+                           },
+                           indent=4,
+                           separators=(',', ':')))
```

Comment:
It feels kind of inconsistent now that we're writing two files with `_write_to_file` but then the third one is being written through `urllib.urlretrieve`. It might make more sense to download the image data into memory and write it with `_write_to_file` so that we're only writing to the filesystem through one API.